src/Features/LanguageServer/CommonLanguageServerProtocol.Framework/ILifeCycleManagement.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace CommonLanguageServerProtocol.Framework;
+
+public interface ILifeCycleManager
```

Comment:
Few questions:
- What's the motivation behind interface's for these instead of abstract classes? 
- Does it makes sense for the lifecycle manager to have a "Initialized" method?
- Is the intention behind the methods here to be notification of shutdown / exit / initialized OR actual logic? Asking because of it was for notification purposes we may want to try `OnShutdown` / `OnExit` etc.

Reply:
Here's the concrete implementations I've got right now. As-is it's a way to abstract out the execution of the corresponding actions on the LanguageServer object.